### PR TITLE
test: Add kubernetes custom dns test

### LINKED ISSUE #1281: Kubernetes custom dns test
Add a test to configure a pod's custom dns

### PR DESCRIPTION
**Pull request by GabyCT**
This will configures a custom pod's dns config.

Fixes #1281

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: integration/kubernetes/k8s-custom-dns.bats [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../.ci/lib.sh"
+load "${BATS_TEST_DIRNAME}/../../lib/common.bash"
+
+setup() {
+	export KUBECONFIG="$HOME/.kube/config"
+	pod_name="custom-dns-test"
+	file="/etc/resolv.conf"
```
Comment (grahamwhaley) [inline integration/kubernetes/k8s-custom-dns.bats]: `file`  is also a command, you might want to call it something else just for clarity :-)

**Comment by grahamwhaley** [comment]
/test
let's see if it flies...

File: integration/kubernetes/k8s-custom-dns.bats [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+@test "Check custom dns" {
+	# Create the pod
+	kubectl create -f "${pod_config_dir}/pod-custom-dns.yaml"
+
+	# Check pod creation
+	kubectl wait --for=condition=Ready pod "$pod_name"
+
+	# Check dns config at /etc/resolv.conf
+	kubectl exec -it "$pod_name" -- cat "$file" | grep "nameserver 1.2.3.4"
```
Comment (jodh-intel) [inline integration/kubernetes/k8s-custom-dns.bats]: Nit: `grep -q`?

**Pull request merged**